smartsim/_core/_install/buildenv.py

```diff
@@ -539,20 +557,26 @@ def check_build_dependency(self, command: str) -> None:
         except OSError:
             raise SetupError(f"{command} must be installed to build SmartSim") from None
 
-    @staticmethod
-    def check_installed(package: str, version: t.Optional[Version_] = None) -> bool:
+    @classmethod
+    def check_installed(cls, package: str, version: t.Optional[str] = None) -> bool:
         """Check if a package is installed. If version is provided, check if
         it's a compatible version. (major and minor the same)"""
         try:
-            installed = Version_(pkg_resources.get_distribution(package).version)
-            if version:
-                # detect if major or minor versions differ
-                if installed.major != version.major or installed.minor != version.minor:
-                    msg = (
-                        f"Incompatible version for {package} detected.\n"
-                        + f"{package} {version} requested but {package} {installed} installed."
-                    )
-                    raise SetupError(msg)
-            return True
-        except pkg_resources.DistributionNotFound:
+            installed = cls.get_package_version(package)
+        except importlib.metadata.PackageNotFoundError:
             return False
+        if version:
+            if not isinstance(version, Version_):
+                version = Version_(version)
+            # detect if major or minor versions differ
+            if installed.major != version.major or installed.minor != version.minor:
+                msg = (
+                    f"Incompatible version for {package} detected: "
```

Comment:
do you need a `+` to concatenate these strings?

Reply:
You do not for string literals or f-strs. No difference in performance either; at the byte code level string concatenation of str literals and f-strs, both with and without  `+`, are optimized to be equivalent.

I simply prefer the no `+` just bc its less to type. I don't think we ever agreed upon a specific style for SS, but I imagine the with `+` is more prevalent. I'd be willing to change if we think it is best to match the existing style!